src/cdk-experimental/ui-patterns/behaviors/event-manager/event-manager.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+/**
+ * An event that supports modifier keys.
+ */
+export interface EventWithModifiers extends Event {
+  ctrlKey: boolean;
+  shiftKey: boolean;
+  altKey: boolean;
+  metaKey: boolean;
+}
+
+/**
+ * Options that are applicable to all event handlers.
+ */
+export interface EventHandlerOptions {
+  stopPropagation: boolean;
+  preventDefault: boolean;
+}
+
+/**
+ * A config that specifies how to handle a particular event.
+ */
+export interface EventHandlerConfig<T extends Event> extends EventHandlerOptions {
```

Comment:
What's the reason for separate `EventHandlerOptions` and `EventHandlerConfig`? They're very similar names for very similar things.

Reply:
Could potentially do a type like `EventHandlerWithConfig` that captures both

Reply:
@mmalerba Would you be able to address this?

Reply:
I think options were like defaults that got applied for every event registered with the handler and config was a configuration for a particular `.on(...)` call. Some of the options do overlap there, e.g. you want to specify that a KeydownHandler defaults to calling preventDefault, but I might create a particular registration where I don't want that behavior, so I can set it on either the default options or the particular `.on` config